xchange-upbit/src/main/java/org/knowm/xchange/upbit/UpbitAdapters.java

```diff
@@ -130,28 +137,30 @@ public static Order adaptOrderInfo(UpbitOrderResponse upbitOrderResponse) {
       status = Order.OrderStatus.PARTIALLY_FILLED;
     } else if (upbitOrderResponse.getState().equalsIgnoreCase("done")) {
       status = Order.OrderStatus.FILLED;
+    } else {
+      status = Order.OrderStatus.NEW;
     }
-    OrderType type = upbitOrderResponse.getSide().equals("ask") ? OrderType.ASK : OrderType.BID;
-    BigDecimal originalAmount = upbitOrderResponse.getVolume();
-    String[] markets = upbitOrderResponse.getMarket().split("-");
-    CurrencyPair currencyPair =
-        new CurrencyPair(new Currency(markets[1]), new Currency(markets[0]));
-    String orderId = upbitOrderResponse.getUuid();
-    BigDecimal cumulativeAmount = upbitOrderResponse.getExecutedVolume();
-    BigDecimal price = upbitOrderResponse.getAvgPrice();
-    ZonedDateTime dateTime = ZonedDateTime.parse(upbitOrderResponse.getCreatedAt());
-    LimitOrder order =
-        new LimitOrder(
-            type,
-            originalAmount,
-            currencyPair,
-            orderId,
-            Date.from(dateTime.toInstant()),
-            price,
-            price,
-            cumulativeAmount,
-            upbitOrderResponse.getPaidFee(),
-            status);
-    return order;
+    return new LimitOrder(
```

Comment:
also sorry missed this but can you use the builder class that limit order has to do this

Reply:
Fixed.